tabby-core/src/services/profiles.service.ts

```diff
@@ -177,30 +307,38 @@ export class ProfilesService {
                     })
                 } catch { }
 
-                this.getProviders().filter(x => x.supportsQuickConnect).forEach(provider => {
-                    options.push({
-                        name: this.translate.instant('Quick connect'),
-                        freeInputPattern: this.translate.instant('Connect to "%s"...'),
-                        description: `(${provider.name.toUpperCase()})`,
-                        icon: 'fas fa-arrow-right',
-                        weight: provider.id !== this.config.store.defaultQuickConnectProvider ? 1 : 0,
-                        callback: query => {
-                            const profile = provider.quickConnect(query)
-                            resolve(profile)
-                        },
-                    })
+                this.getProviders().forEach(provider => {
+                    if (provider instanceof QuickConnectProfileProvider) {
+                        options.push({
+                            name: this.translate.instant('Quick connect'),
+                            freeInputPattern: this.translate.instant('Connect to "%s"...'),
+                            description: `(${provider.name.toUpperCase()})`,
+                            icon: 'fas fa-arrow-right',
+                            weight: provider.id !== this.config.store.defaultQuickConnectProvider ? 1 : 0,
+                            callback: query => {
+                                const profile = provider.quickConnect(query)
+                                resolve(profile)
+                            },
+                        })
+                    }
                 })
 
-                await this.selector.show(this.translate.instant('Select profile or enter an address'), options)
+                await this.selector.show(this.translate.instant('Select profile or enter an address'), options).catch(() => resolve(null))
```

Comment:
same (swallows modal dismissal)

Reply:
Corrected with 49f9a10372d15f7e6b358ba9a43536a8e188e8b7 :
It was a bit dumb of me to resolve the Promise when the selector is dismissed. The Promise is now rejected on selector dismiss.